src/compiler/checker.ts

```diff
@@ -4662,6 +4662,9 @@ namespace ts {
                 // type arguments in the same manner as a type reference to get the same error reporting experience.
                 baseType = getTypeFromClassOrInterfaceReference(baseTypeNode, baseConstructorType.symbol);
             }
+            else if (baseConstructorType.flags & TypeFlags.Any) {
+                baseType = anyType;

```

Comment:
Could you leave a comment saying why it is necessary to set it to `anyType` instead of `baseConstructorType`?